File: CS/GoogleCalendarExample/Log.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GoogleCalendarExample {
    public static class Log {
        static Action<string> logAction;
        public static void Register(Action<string> logDelegate) {
            logAction = logDelegate;
        }

        public static void WriteLine(string message) {
            if (logAction == null)
                return;
            logAction(message);
        }
    }
}

File: VB/GoogleCalendarExample/Log.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace GoogleCalendarExample
	Public NotInheritable Class Log
		Private Shared logAction As Action(Of String)
		Private Sub New()
		End Sub
		Public Shared Sub Register(ByVal logDelegate As Action(Of String))
			logAction = logDelegate
		End Sub

		Public Shared Sub WriteLine(ByVal message As String)
			If logAction Is Nothing Then
				Return
			End If
			logAction(message)
		End Sub
	End Class
End Namespace
